src/Qir/Runtime/lib/QIR/QubitManager.cpp

```diff
@@ -258,93 +257,89 @@ namespace Quantum
         FailIf(qubitsToAllocate == nullptr, "No array provided for qubits to be allocated.");
 
         // Consider optimization for initial allocation of a large array at once
-        for (int32_t i = 0; i < qubitCountToAllocate; i++)
+        for (QubitIdType i = 0; i < qubitCountToAllocate; i++)
         {
             QubitIdType newQubitId = AllocateQubitId();
             if (newQubitId == NoneMarker)
             {
-                for (int32_t k = 0; k < i; k++)
+                for (QubitIdType k = 0; k < i; k++)
                 {
                     Release(qubitsToAllocate[k]);
                 }
                 FailNow("Not enough qubits.");
             }
-            qubitsToAllocate[i] = CreateQubitObject(newQubitId);
+            qubitsToAllocate[i] = newQubitId;
         }
     }
 
-    void CQubitManager::Release(Qubit qubit)
+    void CQubitManager::Release(QubitIdType qubit)
     {
         FailIf(!IsValidQubit(qubit), "Qubit is not valid.");
-        ReleaseQubitId(QubitToId(qubit));
-        DeleteQubitObject(qubit);
+        ReleaseQubitId(qubit);
     }
 
-    void CQubitManager::Release(Qubit* qubitsToRelease, int32_t qubitCountToRelease)
+    void CQubitManager::Release(QubitIdType* qubitsToRelease, QubitIdType qubitCountToRelease)
     {
         if (qubitCountToRelease == 0)
         {
             return;
         }
-        FailIf(qubitCountToRelease < 0, "Cannot release negative number of qubits.");
         FailIf(qubitsToRelease == nullptr, "No array provided with qubits to be released.");
 
-        for (int32_t i = 0; i < qubitCountToRelease; i++)
+        for (QubitIdType i = 0; i < qubitCountToRelease; i++)
         {
             Release(qubitsToRelease[i]);
-            qubitsToRelease[i] = nullptr;
+            qubitsToRelease[i] = 0;
```

Comment:
As the global type is now signed int, a special value such as NoneMarker can be used.